cpp/src/gandiva/regex_util.cc

```diff
@@ -44,6 +45,10 @@ Status RegexUtil::SqlLikePatternToPcre(const std::string& sql_pattern, char esca
 
       cur = sql_pattern.at(idx);
       if (cur == '_' || cur == '%' || cur == escape_char) {
+        if (cur == '\\') {
```

Comment:
Is this condition correct? It seems that we need to check whether the `cur` (== `escape_char`) is a special character or not:

```suggestion
        if (pcre_regex_specials_.find(cur) != pcre_regex_specials_.end()) {
```

BTW, can we extract this condition to out of this `for` loop if my suggestion is correct?

Reply:
This extra condition was needed to escape the '\' in the pcre pattern if the escape character itself is a '\'.

Reply:
This doesn't look correct to me. You should check `cur == escape_char` instead of `curr == '\\'`

@kou in sql like pattern only _, % and escape_char are special characters. And escape_character should only be followed by '_', '%' or the escape char itself as mentioned in the comment above. so we don't need to check for other pcre special characters.